dash/include/dash/HaloMatrix.h

```diff
@@ -0,0 +1,259 @@
+#ifndef DASH__HALOMATRIX_H_INCLUDED
+#define DASH__HALOMATRIX_H_INCLUDED
+
+#include <dash/dart/if/dart.h>
+
+//#include <dash/Team.h>
+#include <dash/Pattern.h>
+//#include <dash/GlobRef.h>
+#include <dash/GlobMem.h>
+//#include <dash/Allocator.h>
+#include <dash/iterator/HaloMatrixIterator.h>
+#include <dash/Halo.h>
+#include <dash/Matrix.h>
+#include <type_traits>
+
+namespace dash {
+
+template<typename MatrixT, typename HaloSpecT>
+class HaloMatrix
```

Comment:
The design of `dash::HaloMatrix` violates a lot of concepts and the container definition.
For example, `ibegin`, `iend` and `bbegin`, `bend` should be views, like `dash::HaloMatrix.boundary.begin|.end`.
Methods `updateHalo`, `waitHaloAsync` etc. are obviously not container concerns, but algorithms.
We would have to re-implement those for every container that supports boundary regions.
I'm okay with putting this implementation in examples, but we can't publish it as a container type.